packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -325,12 +395,41 @@ export class EngineReportingAgent<TContext = any> {
       });
     }
 
+    if (this.options.endpointUrl) {
+      this.logger.warn(
+        '[deprecated] The `endpointUrl` option within `engine` has been renamed to `tracesEndpointUrl`.',
+      );
+    }
+    this.tracesEndpointUrl =
+      (this.options.endpointUrl ||
+        this.options.tracesEndpointUrl ||
+        'https://engine-report.apollodata.com') + '/api/ingress/traces';
+
     // Handle the legacy options: privateVariables and privateHeaders
     handleLegacyOptions(this.options);
   }
 
+  private executableSchemaIdGenerator(schema: string | GraphQLSchema) {
+    if (this.lastSeenExecutableSchemaToId?.executableSchema === schema) {
```

Comment:
This is assuming that (in the GraphQLSchema case) we always get the identical (referentially speaking) object each time. That sounds valid!

I sure hope that the performance impact of comparing two perhaps-large schema strings on every request (in the string case) isn't notable. If it is we can always give more parts of the code access to the hash...